Add notes about wildcard certificates

ptgott opened the pull request:
Guides to getting started with Teleport on various platforms
recommend creating a DNS record for *.teleport.com. It would help
prospective users to know why this is needed. This change adds
context for why Application Access requires a wildcard subdomain.

Fixes #5378

ptgott committed:
Add notes about wildcard certificates

Guides to getting started with Teleport on various platforms
recommend creating a DNS record for *.teleport.com. It would help
prospective users to know why this is needed. This change adds
context for why Application Access requires a wildcard subdomain.

Fixes #5378

zmb3 reviewed (commented):
Whether or not we agree with them, many organizations still see the requirement for a wildcard certificate as less secure or harder to obtain.

It would be nice if we could find a way to suggest wildcard certificates for their ease of use without scaring away folks who don't want to use wildcards.

ptgott committed:
Respond to PR feedback

- Move information into a partial
- Mention that you can create a DNS A record for each application-
  specific subdomain

r0mant commented on docs/pages/kubernetes-access/getting-started/cluster.mdx:
```diff
@@ -131,8 +133,7 @@ for web apps using [application access](../../application-access/introduction.md
     # Create a JSON file changeset for AWS.
     $ jq -n --arg ip ${MYIP?} --arg dns ${MYDNS?} '{"Comment": "Create records", "Changes": [
-      {"Action": "CREATE", "ResourceRecordSet": {"Name": $dns, "Type": "A", "TTL": 300, "ResourceRecords": [{ "Value": $ip}]}},
-      {"Action": "CREATE", "ResourceRecordSet": {"Name": ("*." + $dns), "Type": "A", "TTL": 300, "ResourceRecords": [{ "Value": $ip}]}}
```
I'm not sure why are we removing the commands for creating wildcard records? If we do that, it is no longer a copy-pasteable example :) Same below.

ptgott replied:
Good point—I've restored the original commands while adding a reminder to adjust them for your environment in case, say, a user doesn't want to create a wildcard record.

ptgott commented:
@r0mant Would you have time to give this one another look? Thanks!

ptgott committed:
Restore DNS-related cloud provider commands

ptgott commented:
@zmb3 @r0mant I fixed a lot of conflicts while rebasing this one, so there may be changes to the original commits that require a second look.
